cast/src/lib.rs

```diff
@@ -447,8 +452,21 @@ impl SimpleCast {
         format!("0x{}", s)
     }
 
+    pub async fn generate_interface(address: String, chain: Chain) -> Result<Vec<InterfaceSource>> {
+        let env_key = foundry_utils::etherscan_api_key()?;
```

Comment:
Let's avoid using the helper function, add a new param `etherscan_key: String` to the function and make it callable that way. Makes it easier to use in 3rd party packages in the future.